website/docs/error-reference.md

```diff
@@ -4774,6 +4774,29 @@ result =
 T.reveal_type(result) # => T::Boolean
 ```
 
+## 7052
+
+Occurs when a **positional argument is interpreted as a keyword argument**, but
+has type `T.untyped`. This prevents unintended behavior where **an untyped value
+is mistakenly treated as keyword arguments**, potentially causing runtime
+errors.
+
+You can usually fix this by explicitly using keyword args or ensuring positional
+args have appropriate types.
```

Comment:
```suggestion
has type `T.untyped`. This error is designed to prevent mistaking an untyped
value for keyword arguments, potentially causing runtime errors (e.g., if the
value is not actually a `Hash` at runtime).

The usual fix is to explicitly use keyword args, or to ensure positional args
are typed.
```